Configuration/PyReleaseValidation/python/relval_upgrade.py

```diff
@@ -124,6 +132,6 @@ def makeStepName(key,frag,step,suffix):
                                                       [slist[0]] +                      # Start with signal generation
                                                       stepList['Premix'] +              # Premixing stage1
                                                       [slist[1].replace("PUPRMX", "PUPRMXCombined")] + # Premixing stage2, customized for the combined (defined in relval_steps.py)
-                                                      slist[2:]]                        # Remaining standard steps
+                                                      filter(lambda x: "nano" not in x.lower(), slist[2:])]    # Remaining standard steps (excluding NANOAOD because of input file numbering mismatch)
```

Comment:
The `NanoFull` step is defined here
https://github.com/cms-sw/cmssw/blob/1c42b28cbc052ee06768513b450d9e9fc0f42fd6/Configuration/PyReleaseValidation/python/relval_steps.py#L3199-#L3206
to use `step3_inMINIAODSIM.root` as the input file. In the combined premixing workflow the corresponding file is named as `step4_inMINIAODSIM.root`. I'm not sure what would be the best way to deal with that. Some options coming to mind
1. drop NANOAOD step (as done here); it is not part of any other combined premixing workflow
2. deal with the difference in `relval_steps`, e.g. define programmatically something like `NanoFullPUPRMXCombined` and replace `NanoFull` with that here (like the Digi step is modified above)
3. deal with the difference in `WorkFlowRunner.py` similarly to how HARVESTING input file is handled

Reply:
@makortel I believe we should go in the direction of including more nanoAOD, as they are nowadays an integral part of our production chain. Your option 2 looks a possible solution to me, but may be @prebello @zhenhu @chayanit have other suggestions

Reply:
The last commit keeps Nano step in these workflows with the approach 2.